cypress/integration/vitessce-data-uri.spec.js

```diff
@@ -0,0 +1,89 @@
+
+import { urlPrefix } from '../../src/app/api';
+
+Cypress.on('window:before:load', (win) => {
+  // Forces fallback to XHR, so cypress can mock response.
+  //
+  // Using this work-around:
+  // https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/stubbing-spying__window-fetch
+  //
+  // until this feature is implemented:
+  // https://github.com/cypress-io/cypress/issues/95
+
+  delete win.fetch; // eslint-disable-line no-param-reassign
+});
+
+function loadConfig(config) {
+  // Without a route, Cypress tries to proxy the request,
+  // and that doesn't seem to work for data URIs.
+  const uri = 'data:,JSON-HERE';
+  cy.route({
+    url: uri,
+    response: config,
+  });
+
+  cy.visit('/');
+  cy.get('input[name=url]').type(uri);
+  cy.get('button').click();
+}
+
+describe('Vitessce Data URIs', () => {
+  beforeEach(() => {
+    // Any request we do not explicitly route will return 404,
+    // so we won't end up depending on outside resources by accident.
+    cy.server({ force404: true });
+  });
+
+  it('loads valid data URI', () => {
```

Comment:
Is there a URI here or just an object?

Reply:
As a follow up, are we actually testing success of the data uri then?

Reply:
> Is there a URI here or just an object?

Both: If you take a viewconf json, and prefix it with `data:,` you can paste that in the "Or specify URL of configuration" slot on the homepage.

> are we actually testing success of the data uri then?

I think so: Different viewconfs result in a different state on screen, and each test here has a different set of assertions.

Reply:
I guess my question is more about what we're testing - the data URI is valid because it's the same for all the tests but what changes is the response object.  Do I have this wrong?